MAINT: Start applying ruff/Pycodestyle rules

DimitriPapadopoulos opened the pull request:
Start working on #24994.

I have left out most of `distutils`, `*.pyi` files, and some special cases that seem to require extra attention.

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule F811

F811 Redefinition of unused `...`

I have left out *.pyi files and some special cases.

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule F821

F821 Undefined name

I have left out *.pyi files and some special cases.

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule E701

E701 Multiple statements on one line (colon)

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule F541

F541 f-string without any placeholders

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule E714

E714 Test for object identity should be `is not`

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule E401

E401 Multiple imports on one line

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule E722

E722 Do not use bare `except`

eendebakpt commented on numpy/_core/arrayprint.py:
```diff
@@ -85,7 +85,7 @@ def _make_options_dict(precision=None, threshold=None, edgeitems=None,
         raise ValueError("sign option must be one of ' ', '+', or '-'")
 
-    if legacy == False:
+    if legacy is False:
```
This changes behaviour for `legacy` equal to `O`. Maybe it is ok in this case, but I am a bit surprised the change end up here

DimitriPapadopoulos replied:
Changing `x == False` → `x is False` works when the `x` is a `bool`. I am not certain automatic fixes check that. In any case, comparing a non-`bool` to `False` is poor style that needs to be fixed.

DimitriPapadopoulos replied:
I'll leave out such fixes for now.

DimitriPapadopoulos replied:
For [numpy.array](https://numpy.org/doc/stable/reference/generated/numpy.array.html), `== None` and `is None` are not equivalent. Therefore, the ruff/pycodestyle rule [E711](https://docs.astral.sh/ruff/rules/none-comparison/) cannot be applied.

mattip replied:
Did you mean to revert this change?

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule E713

E713 Test for membership should be `not in`

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule F401

F401 imported but unused

I have left out *.pyi files. Still lots of occurrences to examine.

Co-authored-by: Robert Kern <robert.kern@gmail.com>

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule E711

E711 Comparison to `None` should be `cond is None`

DimitriPapadopoulos committed:
MAINT: apply ruff/Pycodestyle rule E712

E712 Avoid inequality comparisons to `True`; use `if not cond:` for false checks

rkern commented on numpy/_core/arrayprint.py:
```diff
@@ -85,7 +85,13 @@ def _make_options_dict(precision=None, threshold=None, edgeitems=None,
+        options['legacy'] = sys.maxsize
+    elif legacy == False:  # noqa: E712
+        warnings.warn(
+            f"Passing `legacy={legacy}` is deprecated.",
```
```suggestion
            f"Passing `legacy={legacy!r}` is deprecated.",
```

Just in case. It's almost certainly going to be `0`, but who knows?

rkern commented on numpy/_core/numerictypes.py:
```diff
@@ -95,9 +95,6 @@
 
 # we don't need all these imports, but we need to keep them for compatibility
 # for users using np._core.numerictypes.UPPER_TABLE
```
The comment specifies why these need to remain.

DimitriPapadopoulos replied:
I seem to recall the F401 removals were automatic, then manually reviewed. Some fell through the cracks, I'll fix that. Is it OK to add `# noqa: F401` to these lines?

rkern commented on numpy/_core/numerictypes.py:
```diff
@@ -106,7 +103,7 @@
 # we don't export these for import *, but we do want them accessible
 # as numerictypes.bool, etc.
-from builtins import bool, int, float, complex, object, str, bytes
+from builtins import bool, int, float, complex, str, bytes
```
Note the comment.

rkern commented on numpy/testing/overrides.py:
```diff
@@ -63,7 +63,6 @@ def get_overridable_numpy_array_functions():
     """
     # 'import numpy' doesn't import recfunctions, so make sure it's imported
     # so ufuncs defined there show up in the ufunc listing
-    from numpy.lib import recfunctions
```
Leave this, per the comment.

rkern commented on numpy/tests/test_lazyloading.py:
```diff
@@ -27,7 +26,6 @@ def test_lazy_load():
         np = module
 
         # test a subpackage import
-        from numpy.lib import recfunctions
```
Leave this, per the comment.

rkern reviewed (commented):
I haven't individually confirmed all of the `import` removals, but I did note the ones that had a comment explaining why they were there and should not be removed.

I think the other changes look good to my eyeballs.

DimitriPapadopoulos committed:
Update numpy/_core/arrayprint.py

Co-authored-by: Robert Kern <robert.kern@gmail.com>

mattip commented on numpy/_core/tests/test_umath.py:
```diff
@@ -694,11 +693,11 @@ def test_floor_division_corner_cases(self, dtype):
+            assert(np.isnan(div)), "div: %s" % div
             div = np.floor_divide(fnan, fzer)
-            assert(np.isnan(div)), "dt: %s, div: %s" % (dt, div)
+            assert(np.isnan(div)), "div: %s" % div
```
Is there a reason to not remove the parentheses from the asserts?
```suggestion
            assert np.isnan(div), "div: %s" % div
```

DimitriPapadopoulos replied:
These commits contain automated fixes hen possible. I try not to manually modify these fixes, unless really needed. There's a ruff/pyupgrade rule for extraneous parentheses ([`UP034`](https://docs.astral.sh/ruff/rules/extraneous-parentheses/)), which I could apply in a different PR.

mattip commented on numpy/f2py/_src_pyf.py:
```diff
@@ -1,3 +1,4 @@
+import os
```
Hmm, how did this work before?

DimitriPapadopoulos replied:
Perhaps `resolve_includes`() is usually not called.

mattip commented on numpy/lib/tests/test_arraysetops.py:
```diff
@@ -269,8 +269,6 @@ def assert_isin_equal(a, b):
             assert_isin_equal(empty_array, empty_array)
 
-    @pytest.mark.parametrize("kind", [None, "sort", "table"])
-    def test_isin(self, kind):
```
Rather than remove this, please rename the tests and keep them separate.

mattip commented on numpy/linalg/lapack_lite/clapack_scrub.py:
```diff
@@ -305,7 +305,7 @@ def scrubSource(source, nsteps=None, verbose=False):
         nsteps = None
 
-    source = scrub_source(source, nsteps, verbose=True)
+    source = scrubSource(source, nsteps, verbose=True)
```
Huh. How did this work?

DimitriPapadopoulos replied:
I guess this Python 2.7 script hasn't been used for a long time - probably not after f589305da2758f28eacc33fe0e50f1c1460a4c5e in 2018.

mattip reviewed (commented):
We usually do not like these kinds of code reformats, but this one is actually quite nice. I thing we shouldn't be touching the deprecated numpy.distutils, could you revert those?

charris commented:
Needs rebase.

charris commented:
Thanks @DimitriPapadopoulos .
